vuu-ui/packages/vuu-layout/test/layout-persistence/LocalLayoutPersistenceManager.test.ts

```diff
@@ -0,0 +1,535 @@
+};
+
+const existingLayout: Layout = {
+  id: existingId,
+  json: { type: "t0" }
+};
+
+const metadataToAdd: Omit<LayoutMetadata, "id"> = {
+  name: "New Layout",
+  group: "Group 1",
+  screenshot: "screenshot",
+  user: "vuu user",
+  date: "26/09/2023",
+};
+
+const layoutToAdd: LayoutJSON = {
+  type: "t",
+};
+
+const metadataSaveLocation = "layouts/metadata";
+const layoutsSaveLocation = "layouts/layouts";
+
+afterEach(() => {
+  localStorage.clear();
+})
+
+describe("createLayout", () => {
+
+  it("persists to local storage with a unique ID", async () => {
+    const returnedId = await persistenceManager.createLayout(metadataToAdd, layoutToAdd);
+
+    const persistedMetadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+    const persistedLayout = getLocalEntity<Layout[]>(layoutsSaveLocation);
+
+    const expectedMetadata: LayoutMetadata = {
+      ...metadataToAdd,
+      id: returnedId
+    };
+
+    const expectedLayout: Layout = {
+      json: layoutToAdd,
+      id: returnedId
+    };
+
+    expect(persistedMetadata).toEqual([expectedMetadata]);
+    expect(persistedLayout).toEqual([expectedLayout]);
+  });
+
+  it("adds to existing storage", async () => {
+    saveLocalEntity(metadataSaveLocation, [existingMetadata]);
+    saveLocalEntity(layoutsSaveLocation, [existingLayout]);
+
+    const returnedId = await persistenceManager.createLayout(metadataToAdd, layoutToAdd);
+    expect(returnedId).not.toEqual(existingId);
+
+    const persistedMetadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+    const persistedLayout = getLocalEntity<Layout[]>(layoutsSaveLocation);
+
+    const expectedMetadata: LayoutMetadata = {
+      ...metadataToAdd,
+      id: returnedId,
+    };
+
+    const expectedLayout: Layout = {
+      json: layoutToAdd,
+      id: returnedId,
+    };
+
+    expect(persistedMetadata).toEqual([existingMetadata, expectedMetadata]);
+    expect(persistedLayout).toEqual([existingLayout, expectedLayout]);
+  });
+});
+
+describe("updateLayout", () => {
+
+  it("updates an existing layout", async () => {
+    saveLocalEntity(metadataSaveLocation, [existingMetadata]);
+    saveLocalEntity(layoutsSaveLocation, [existingLayout]);
+
+    await persistenceManager.updateLayout(existingId, metadataToAdd, layoutToAdd);
+
+    const persistedMetadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+    const persistedLayout = getLocalEntity<Layout[]>(layoutsSaveLocation);
+
+    const expectedMetadata: LayoutMetadata = {
+      ...metadataToAdd,
+      id: existingId,
+    };
+
+    const expectedLayout: Layout = {
+      json: layoutToAdd,
+      id: existingId,
+    };
+
+    expect(persistedMetadata).toEqual([expectedMetadata]);
+    expect(persistedLayout).toEqual([expectedLayout]);
+  });
+
+  it("errors if there is no metadata in local storage with requested ID ", async () => {
+    saveLocalEntity(layoutsSaveLocation, [existingLayout]);
+
+    var exceptionCaught = false;
```

Comment:
All `var`s should be `let`s or `const`s.

Reply:
7c380ec1